addons/mail/wizard/mail_compose_message.py

```diff
@@ -1237,9 +1237,10 @@ def _prepare_mail_values_dynamic(self, res_ids):
                     'model': self.model,
                     'res_id': res_id,
                 }
-                message_inmem = self.env['mail.message'].new({
-                    'body': mail_values['body'],
-                })
+                new_mail_message_values = {'body': mail_values['body']}
+                if self.template_id:
+                    new_mail_message_values.update({'email_add_signature': True})
```

Comment:
No need of update for a single key actually
```suggestion
                    new_mail_message_values['email_add_signature'] = True
```